specs/marketplace.md

```diff
@@ -0,0 +1,374 @@
+  byte32 merkleRoot;
+}
+
+```
+
+Notes about some parameters:
+
+`cid` 
+
+An identifier used to locate the Manifest representing the dataset.
+- MUST be a [CIDv1](https://github.com/multiformats/cid#cidv1) with sha-256 based [multihash](https://github.com/multiformats/multihash).
+- Data it represents SHOULD be discoverable in the network, otherwise the Request will get canceled. 
+
+`reward`
+
+- It MUST be an amount of tokens offered per slot per second.
+- The Ethereum address that submits the `requestStorage()` transaction MUST have [approval](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#IERC20-approve-address-uint256-) for transfer of at least the same amount on the ERC20 based token, that the network uses. 
+
+`collateral`
+
+- Amount of tokens that the storage providers submit when they fill slots.
+- Collateral is then slashed or forfeited if the storage providers fail to provide the service requested by the Request (more information bellow).
+
+`proofProbability`
+
+Determines the inverse probability that a proof is required in a period: $\frac{1}{proofProbability}$
+
+- Storage providers are required to provide proofs of storage to the marketplace smart contract when they are prompted to by the smart contract.
+- The frequency is stochastic in order to prevent hosts from only coming online when proofs are required, but it is affected by this parameter.
+
+`expiry`
+
+- The parameter is specified as duration in seconds, hence the final deadline timestamp is calculated at the moment when the transaction is mined. 
+
+`nonce`
+
+- It SHOULD be a random byte array.
+
+#### Renewal of Storage Request
+
+It should be noted that Marketplace does not support extending Requests. It is REQUIRED that if the user wants to 
+extend the Request's duration, somebody submits a new Request transaction with the same CID **well before the original
+Request finishes**. In this way, the data will be still persisted in the network at the time when new (or the current) storage providers 
+need to retrieve the dataset to fill slots of the new Request.
+
+### Withdrawing funds
+
+The client node SHOULD monitor the status of Requests that it created. When Request reaches states `Cancelled` (not all slots filled after `expiry` timeout)
+the client node SHOULD initiate withdrawal of the remaining funds from the contract using function `withdrawFunds(requestId)`.
+
+ - `Cancelled` state MAY be detected using timeout specified from function `requestExpiresAt(requestId)` **and** not detecting emitted `RequestFulfilled(requestId)` event.
+ - `Failed` state MAY be detected using `RequestFailed(requestId)` event emitted from the smart contract.
+ - `Finished` state MAY be detected by setting timeout specified from function `getRequestEnd(requestId)`.
+
+## Storage Provider role
+
+Storage Provider (SP) role represents nodes that persist data across the network by hosting Slots of Requests 
+that Client nodes requested.
```

Comment:
I know, but this is the start of the chapter that discuss the SPs, so I would like to have it "redefined" here just in case somebody jumps directly to here and then he wonders what "SP" means ;-)

Reply:
```suggestion
Storage Provider (SP) role represents nodes that persist data across the network by hosting Slots of Requests 
that the Client nodes requested.
```